pkg/k8s/service_cache.go

```diff
@@ -243,17 +247,18 @@ func (s *ServiceCache) GetEndpointsOfService(svcID ServiceID) *Endpoints {
 	return eps.GetEndpoints()
 }
 
-// GetNodeAddressing returns the registered node addresses to this service cache.
-func (s *ServiceCache) GetNodeAddressing() types.NodeAddressing {
-	return s.nodeAddressing
-}
-
 // UpdateService parses a Kubernetes service and adds or updates it in the
 // ServiceCache. Returns the ServiceID unless the Kubernetes service could not
 // be parsed and a bool to indicate whether the service was changed in the
 // cache or not.
 func (s *ServiceCache) UpdateService(k8sSvc *slim_corev1.Service, swg *lock.StoppableWaitGroup) ServiceID {
-	svcID, newService := ParseService(k8sSvc, s.nodeAddressing)
+	var addrs []netip.Addr
+	if s.nodeAddrs != nil {
+		iter, _ := s.nodeAddrs.Get(s.db.ReadTxn(), datapathTables.NodeAddressNodePortIndex.Query(true))
```

Comment:
The reconciliation when these change is currently performed in `pkg/service/reconciler.go`. It's still racy since `ParseService` may use stale data, which is why the reconciler runs periodically even when nothing changes. Longer term we'll want to remove this query from here and push this expansion down the stack.